test/statetest/statetest_loader.cpp

```diff
@@ -264,12 +264,24 @@ state::Transaction from_json<state::Transaction>(const json::json& j)
 {
     state::Transaction o;
     from_json_tx_common(j, o);
+    if (const auto chain_id_it = j.find("chainId"); chain_id_it != j.end())
```

Comment:
Where is this optional?

Reply:
Ok it's in our code base in `precompile.hpp.` I will move it to better place in separated PR and merge it before this one.

Reply:
When generating state tests using `retesteth` `txs.json` does not contain `chainId`. Probably because it's passed via program param.
```
Running tests for config 'Ethereum GO on StateTool' 2
Test Case "stCallCodes": (1 of 67)
Finishing retesteth run
*** Total Tests Run: 0

Error: [json.exception.out_of_range.403] key 'chainId' not found
 (GeneralStateTests/stCallCodes/callcodecallcall_100, fork: Berlin, TrInfo: d: 0, g: 0, v: 0)
Error: ERROR OCCURED FILLING TESTS: [json.exception.out_of_range.403] key 'chainId' not found
 (GeneralStateTests/stCallCodes/callcodecallcall_100, fork: Berlin, TrInfo: d: 0, g: 0, v: 0)

--------
TestOutputHelper detected 2 errors during test execution!
/Users/rodia/projects/retesteth/retesteth/TestOutputHelper.cpp:226: error: in "GeneralStateTests/stCallCodes":
```

Reply:
I'm fine with whatever. I was just curious.